machinery/gateway_api_types.go

```diff
@@ -315,6 +499,82 @@ func (t LocalPolicyTargetReferenceWithSectionName) GetName() string {
 	return namespacedSectionName(string(t.LocalPolicyTargetReference.Name), *t.SectionName)
 }
 
+// These are wrappers for Gateway API types so instances can be used as objects in the topology.
+
+type ReferenceGrant struct {
```

Comment:
Could the lib provide link func helpers?

Reply:
I've been thinking on this but I'm a bit confused what kind of links funcs for `ReferenceGrants` to do 😅 

Are we linking everything under `referencegrant.spec.from` and `referencegrant.spec.to` to the `RefernceGrant` or are we just linking `referencegrant.spec.from` to `referencegrant.spec.to` or are you thinking something else? 🤔

Reply:
We've decided ReferanceGrant linking to a seperate PR